src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -195,6 +201,32 @@ public class CpsScmFlowDefinitionTest {
         r.assertLogContains("version one", b);
     }
 
+    @Test public void lightweight_brach_parametrised() throws Exception {
+        LoggerRule lr = new LoggerRule();
+        lr.record(GitSCMFileSystem.class.getName(), Level.ALL).capture(4024);
+
+        sampleRepo.init();
+        sampleRepo.git("checkout","-b","master2");
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.git("commit", "--message=init");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("BRANCH","","")));
+        GitStep step = new GitStep(sampleRepo.toString());
+        step.setBranch("${BRANCH}");
+
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(step.createSCM(), "flow.groovy");
+        def.setLightweight(true);
+        TestDurabilityHintProvider provider = Jenkins.get().getExtensionList(TestDurabilityHintProvider.class).get(0);
+        provider.registerHint("p", FlowDurabilityHint.PERFORMANCE_OPTIMIZED);
```

Comment:
```suggestion
```

Irrelevant here?

Reply:
I was wondering if the test shouldnt run faster, but ur right

Reply:
done